### PR TITLE
Refactor for "get_experiment_by_name" and sphinx markup in docs.

### PR DESCRIPTION
**Pull request by mparkhe**
addressing comments in PR #889

- Basic implementation of `get_experiment_by_name` to base class, SQL has a specialized implementation.
- Added sphinx markup for `AbstractStore` for pydocs

### CODE REVIEW

File: mlflow/store/rest_store.py [hunk 1]
```diff
@@ -93,18 +93,13 @@ def get_experiment(self, experiment_id):
...
         :param experiment_id: Integer id for the experiment
-        :return: A single Experiment object if it exists, otherwise raises an Exception.
+        :return: A single :py:class:`mlflow.entities.Experiment` object if it exists,
+        otherwise raises an Exception.
         """
         req_body = message_to_json(GetExperiment(experiment_id=experiment_id))
         response_proto = self._call_endpoint(GetExperiment, req_body)
         return Experiment.from_proto(response_proto.experiment)
 
-    def get_experiment_by_name(self, experiment_name):
```
Comment (mparkhe) [inline mlflow/store/rest_store.py]: moved to base class.

File: mlflow/store/file_store.py [hunk 2]
```diff
@@ -220,13 +220,6 @@ def get_experiment(self, experiment_id):
                                   databricks_pb2.RESOURCE_DOES_NOT_EXIST)
         return experiment
 
-    def get_experiment_by_name(self, experiment_name):
```
Comment (mparkhe) [inline mlflow/store/file_store.py]: same as base class implementation.

File: mlflow/store/abstract_store.py [hunk 3]
```diff
@@ -45,19 +45,24 @@ def get_experiment(self, experiment_id):
         Fetches the experiment by ID from the backend store.
 
         :param experiment_id: Integer id for the experiment
-        :return: A single Experiment object if it exists, otherwise None.
+        :return: A single :py:class:`mlflow.entities.Experiment` object if it exists,
+        otherwise raises an exception.
```
Comment (andrewmchen) [inline mlflow/store/abstract_store.py]: nit: Could we indent newlines like https://github.com/kennethreitz/requests/blob/master/requests/api.py?

**Review by andrewmchen — COMMENTED** [review]
1 nit only.

**Pull request merged**